aten/src/ATen/native/sparse/ValidateCompressedIndicesCommon.h

```diff
@@ -0,0 +1,285 @@
+#pragma once
+
+#include <ATen/Tensor.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/Dispatch.h>
+
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/Functions.h>
+#include <ATen/NativeFunctions.h>
+#else
+#include <ATen/ops/arange.h>
+#include <ATen/ops/empty.h>
+#include <ATen/ops/_sparse_coo_tensor_with_dims_and_tensors.h>
+#endif
+
+#if defined(__CUDACC__) || defined(__HIPCC__)
+#define GPUCC
+#define FUNCAPI __host__ __device__
+#define INLINE __forceinline__
+#define NAME "compressed_index_invariance_checks_cuda"
+#else
+#define FUNCAPI
+#define INLINE inline
+#define NAME "compressed_index_invariance_checks_cpu"
+#endif
+
+#if defined(_WIN32) || defined(_WIN64)
+#define RESTRICT __restrict
+#else
+#define RESTRICT __restrict__
```

Comment:
IMO it would make sense to add CUDA versions to some of those conditional defines. But that might be out-of-scope for this PR, something to think about for the future though.

Reply:
Agree, it might be I am using conditions that might not work all the time.